### DISCUSSION TITLE
Struggling with ESP32-S2

### DISCUSSION THREAD

**Discussion by Tobidogger** [khoih-prog/ESPAsync_WiFiManager#94] [Q&A] [answered]
My top line question is has anyone been able to get one of the examples to work on the adafruit QT PY ESP32-S2, adafruit  ESP32-S2 or adafruit ESP32-S2-TFT? I have have perfect success getting examples (and my changes) to run on a variety of other ESP32's and ESP8266's, but the three previously mentioned adafruit board all end in the same spot:

ESPAsync_WiFiManager v1.12.2
SPIFFS/LittleFS failed! Already tried formatting.
LittleFS failed!. Please use SPIFFS or EEPROM. Stay forever

I have loaded LittleFS and SPIFFS tests on all of these boards, so I know they work.

ESP 2.0.3, ESPAsync_WiFiManager v1.12.2, LittleFS_esp32 1.0.6, ESP_DoubleResetDetector 1.3.1, ArduinoJSON 6.19.4

What might I be missing in my setup? Is there something peculiar to the adafruit boards? Some "I've been staring at this too long" thing in my setup :)

THANKS!

**Comment by khoih-prog** [comment, marked as answer]
Hi @Tobidogger 

Check [**ESP32-S2 ESPAsync_WiFiManager & CircuitPython Alternative**](https://forums.adafruit.com/viewtopic.php?f=57&t=190911#p926838)

It seems there is some issues with **Adafruit** ESP32-S2 (Feather S2, Feather S2 TFT, Metro S2, Funhouse, Magtag, QT PY S2, etc.) firmware, and you'll have to wait for PR [**update adafruit boards tinyuf2 and partition scheme** #6770](https://github.com/espressif/arduino-esp32/pull/6770).

I'm sorry I don't have those boards to know what's actually wrong with them. Post the issue on Adafruit Forum for help.

Good Luck,

**Reply by Tobidogger** [reply]
Ok thanks for the heads up and help.

If you would like a couple of them, I have a few TFT and QT PY S2's I'd be happy to share along as a donation. Or I will just "buy you coffee" as a thanks!

I'll mark this as answer and leave it here for anyone else that comes along.

**Comment by Tobidogger** [comment]
My other account is blocked here for some reason...but this is what I figured out, leaving this here for others:

```
[env:featheresp32-s2]
board_build.flash_mode = qio
platform = https://github.com/platformio/platform-espressif32.git#master
board = featheresp32-s2
framework = arduino
platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#idf-release/v4.4
lib_deps = 
	https://github.com/khoih-prog/ESP_DoubleResetDetector.git
	https://github.com/khoih-prog/ESPAsync_WiFiManager.git
	https://github.com/me-no-dev/ESPAsyncWebServer.git
	https://github.com/me-no-dev/ESPAsyncTCP.git
	https://github.com/me-no-dev/AsyncTCP.git
	bblanchon/ArduinoJson@^6.19.4
	adafruit/Adafruit MQTT Library@^2.4.2
	adafruit/Adafruit BME280 Library@^2.2.2
build_flags = 
	-D DEBUG_ESP_PORT=Serial
	-D NDEBUG
upload_speed = 921600
monitor_speed = 115200
lib_ignore = 
	WiFi101
	lorol/LittleFS_esp32
	LittleFS_esp32
build_type = debug
```

The key seems to be the "board_build.flash_mode = qio"

Core 6.0.2·Home 3.4.2 for platformio
code: Version: 1.68.1 

working on all my adafruit esp32-s2's (feather, tft, qt specifically)
